scan-command/src/main/java/io/ballerina/scan/internal/ProjectAnalyzer.java

```diff
@@ -113,7 +113,8 @@ List<Issue> analyze(List<Rule> inbuiltRules) {
     private Consumer<DocumentId> analyzeDocument(Module module, ScannerContextImpl scannerContext) {
         return documentId -> {
             Document document = module.document(documentId);
-            StaticCodeAnalyzer analyzer = new StaticCodeAnalyzer(document, scannerContext);
+            StaticCodeAnalyzer analyzer = new StaticCodeAnalyzer(document, scannerContext,
+                    module.getCompilation().getSemanticModel());
```

Comment:
Can't you extract this out outside the consumer declaration?

Reply:
For this we need to get the document using the document id.
If we extract it outside we are not able to get the document id

Reply:
I meant `module.getCompilation().getSemanticModel()`.